### PR TITLE
build: move version-info into checksum file

### PR DESCRIPTION
**Pull request by holiman**
This PR builds on top of https://github.com/ethereum/go-ethereum/pull/28316, moving the version-metadata into the checksum-file. 

This makes it so that we only need to update one place when updating go, golang-ci or the execution spec tests.

### CODE REVIEW

File: build/ci.go [hunk 1]
```diff
@@ -149,10 +144,6 @@ var (
 	// versions.
 	gobootVersion = "1.19.6"
```
Comment (karalabe) [inline build/ci.go:145]: You forgot to also use the version for this from the checksums. Or was i deliberate?

Comment (holiman) [reply build/ci.go:145]: It was not deliberate, no

Comment (holiman) [reply build/ci.go]: Fixed, ptal

**Review by karalabe — APPROVED** [review]
SGTM

**Pull request merged**